File: csharpapps/KiranaAppv1/KiranaAppV1.Core/DTOs/Responses/ApiResponse.cs
using FluentValidation.Results;

namespace KiranaAppV1.Core.DTOs.Responses;

public class ApiResponse<T>
{
    public bool Success {get; set;}
    public string Message {get; set;}
    public T Data {get; set;}
    public List<string> Errors {get; set;}

    public ApiResponse(){}
    
    public ApiResponse(bool success, string message, T data, List<string> errors)
    {
        Success = success;
        Message = message;
        Data = data;
        Errors = errors;
    }
}

File: springapps/addressbookv3/src/main/java/com/cloudthat/addressbookv3/dtos/ApiResponse.java
package com.cloudthat.addressbookv3.dtos;

public record ApiResponse<T>(
        Boolean success,
        String message,
        T data,
        Long timestamp
) {
    public ApiResponse{
        if(timestamp == 0){
            timestamp = System.currentTimeMillis();
        }

    }
}
